### ISSUE TITLE
@EachNotEmpty cannot find validator with hibernate-validator 5.1.3

### ISSUE THREAD

**Issue by mbyrne00** [jirutka/validator-collection#12]
Using `hibernate-validator` 5.1.3 (version packaged with latest spring-boot 1.2.3) and version 2.1.3 of this library whenever I use the `@EachNotEmpty` validator I get the following exception ...

```
java.lang.IllegalArgumentException: No validator found for constraint: interface org.hibernate.validator.constraints.NotEmpty
	at org.apache.commons.lang3.Validate.notEmpty(Validate.java:349)
	at cz.jirutka.validator.collection.CommonEachValidator.initialize(CommonEachValidator.java:124)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.initializeConstraint(ConstraintValidatorManager.java:245)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.createAndInitializeValidator(ConstraintValidatorManager.java:147)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorManager.getInitializedValidator(ConstraintValidatorManager.java:101)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:125)
	at org.hibernate.validator.internal.engine.constraintvalidation.ConstraintTree.validateConstraints(ConstraintTree.java:91)
	at org.hibernate.validator.internal.metadata.core.MetaConstraint.validateConstraint(MetaConstraint.java:83)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraint(ValidatorImpl.java:547)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForDefaultGroup(ValidatorImpl.java:487)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateConstraintsForCurrentGroup(ValidatorImpl.java:451)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:403)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraint(ValidatorImpl.java:723)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraints(ValidatorImpl.java:617)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:412)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraint(ValidatorImpl.java:723)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateCascadedConstraints(ValidatorImpl.java:601)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validateInContext(ValidatorImpl.java:412)
	at org.hibernate.validator.internal.engine.ValidatorImpl.validate(ValidatorImpl.java:206)
	at org.springframework.validation.beanvalidation.SpringValidatorAdapter.validate(SpringValidatorAdapter.java:108)
	at org.springframework.validation.DataBinder.validate(DataBinder.java:781)
```

I had a look inside the `hibernate-validator` jar and within the package `org.hibernate.validator.internal.constraintvalidator` I can see a `NotNullValidator` but cannot see one for `NotEmpty` so they must be using a different strategy.

Only workaround I can see for now is to use a different annotation such as `@EachNotNull` for now and that works.

**Comment by And390** [comment]
I have the same error.

**Comment by UtopianStorm** [comment]
The last line of the initialiser method in his ConstraintValidator's validates whether or not there were are validators associated with the hierarchy of the Constraint-annotation being processed.

Since NotEmpty is implemented using NotNull and Size(min = 1); both not explicitly declaring any validators, no validators will be picked up and thus it will fail with that validation error.

A workaround could be implementing NotEmpty using a validator, but yuck; I do not want to write a ConstraintValidator for Sets, Maps, Lists, Collections in general and Arrays not to mention Strings.
I was already crying when I wrote a NaN and NotNaN ConstraintValidator for Float and Double (out of fear of primitives/box-types would be a real bother, although fortunately they didn't cause any trouble).

I would rather see a ConstraintValidator that could apply the Constraints that do not specify a Validator; that would be great.
But kudos to Jirutka anyway.

**Comment by UtopianStorm** [comment]
It's quite ironic though isn't it, that the simplest annotation to implement (NotEmpty - since it does't even require a validator) becomes the most troublesome Constraint using this framework.
I find it bothersome too that the provided annotations haven't been tested, my guess is EachNotEmpty would always have thrown an exception and as such it is not merely a bug introduced through regression.

**Comment by jirutka** [comment]
Sorry for late delay and thanks to @UtopianStorm finding the cause of this issue, it helped me to resolve it more quickly.

I’ve added basic integration tests for all the @EachX annotations (see [here](https://github.com/jirutka/validator-collection/blob/6270acef7c540f61a7426429b5f0a898f385bc87/src/test/groovy/cz/jirutka/validator/collection/constraints/EachAnnotationIT.groovy#L101-L150)), fixed all dysfunctional annotations (EachNotEmpty, EachRange, EachCreditCardNumber) and released v2.1.4 (it’s gonna be in the Maven Central in few hours).
